### PR TITLE
Use CDB author and name in installed package's config

### LINKED ISSUE #8269: Some installed games not detected as installed
##### Issue type
 - Bug report

##### Minetest version
```
$ bin/minetest --version
Minetest 5.0.0-dev-0ad96cc7 (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="."
```

##### OS / Hardware
Operating system: Gentoo Linux

##### Summary
Games installed trough content db are not detected as installed.

##### Steps to reproduce
1. Fetch minetest_game or whynot_game from contentdb
2. Button remain "Install" because not detected as installed

I looked deeper to the lua code

The entry in `store.packages_full` is
```
	{
		thumbnail = "https://content.minetest.net/thumbnails/1/Q4Tym9Rddd.png",
		id = "bell07/whynot_game",
		url = "https://content.minetest.net/packages/bell07/whynot_game/releases/1017/download/",
		score = 3417.9,
		title = "Whynot",
		short_description = "Aims to get all existing high quality mods working together",
		type = "game",
		release = 1017,
		name = "whynot_game",
		author = "bell07"
	},
```
The entry in `game_hash` is

```
	["bell07/whynot"] = {
		gamemods_path = "/home/user/Minetest-5/bin/../games/whynot_game/mods",
		id = "whynot",
		menuicon_path = "/home/user/Minetest-5/bin/../games/whynot_game/menu/icon.png",
		path = "/home/user/Minetest-5/bin/../games/whynot_game",
		addon_mods_paths = {
			"/home/user/Minetest-5/bin/../mods"
		},
		type = "game",
		release = 1017,
		name = "Why not subgame",
		author = "bell07"
	},
```

The "bell07/whynot_game" does not match "bell07/whynot". The game was installed trough the content manager

### PR DESCRIPTION
**Pull request by pauloue**
Fixes #8269 (specifically the issue with MTG not being detected because of a different author in the game.conf. Preinstalled games without an author in the game.conf won't be detected, but that probably can't be fixed since there's no guarantee that it's the same game as the one on CDB.)

### CODE REVIEW

**Comment by rubenwardy** [comment]
Authors should be compared case insensitively, maybe I'm forgetting a :lower()

**Comment by pauloue** [comment]
OK, did that.

File: builtin/mainmenu/dlg_contentstore.lua [hunk 1]
```diff
@@ -317,22 +314,22 @@ function store.update_paths()
...
+			mod_hash[mod.author:lower() .. "/" .. mod.name] = mod
 		end
 	end
 
 	local game_hash = {}
 	pkgmgr.update_gamelist()
 	for _, game in pairs(pkgmgr.games) do
-		if game.author then
-			game_hash[game.author .. "/" .. game.id] = game
+		if game.author ~= "" then
```
Comment (rubenwardy) [inline builtin/mainmenu/dlg_contentstore.lua]: I don't think this should work. `author` should either be filled or nil, but never an empty string

Comment (pauloue) [reply builtin/mainmenu/dlg_contentstore.lua]: Actually the tables returned by `get_games` always have the author field
https://github.com/minetest/minetest/blob/master/src/script/lua_api/l_mainmenu.cpp#L434

**Pull request merged**